pypit/arproc.py

```diff
@@ -218,16 +220,18 @@ def background_subtraction(slf, sciframe, varframe, k=3, crsigma=20.0, maskval=-
     return sciframe-skybg, skybg
 
 
-def badpix(slf, det, frame, sigdev=10.0):
+def badpix(det, frame, sigdev=10.0):
```

Comment:
Good point. This probably needs to be decided on a case by case basis. For example, a 'weights' keyword might be used by one function call and not another, and I don't see the need to put this variable in a settings file. It's certainly something to keep in mind going forward.